src/Ombi.TheMovieDbApi/TheMovieDbApi.cs

```diff
@@ -99,65 +105,95 @@ public async Task<List<MovieSearchResult>> SimilarMovies(int movieId, string lan
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("language", langageCode);
+            var request = new Request($"discover/movie", BaseUri, HttpMethod.Get);
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("language", langCode);
+            var settings = await Settings;
+            request.AddQueryString("include_adult", settings.ShowAdultMovies.ToString().ToLower());
+            request.AddQueryString("without_keywords", settings.ExcludedKeywordIds);
+
             AddRetry(request);
             var result = await Api.Request<TheMovieDbContainer<SearchResult>>(request);
             return Mapper.Map<List<MovieSearchResult>>(result.results);
         }
 
-        public async Task<List<MovieSearchResult>> TopRated(string langageCode)
+        public async Task<List<MovieSearchResult>> TopRated(string langCode)
         {
-            var request = new Request($"movie/top_rated", BaseUri, HttpMethod.Get);
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("language", langageCode);
+            var request = new Request($"discover/movie", BaseUri, HttpMethod.Get);
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("language", langCode);
+            request.AddQueryString("vote_count.gte", "250");
+            request.AddQueryString("sort_by", "vote_average.desc");
+            var settings = await Settings;
+            request.AddQueryString("include_adult", settings.ShowAdultMovies.ToString().ToLower());
+            request.AddQueryString("without_keywords", settings.ExcludedKeywordIds);
+
             AddRetry(request);
             var result = await Api.Request<TheMovieDbContainer<SearchResult>>(request);
             return Mapper.Map<List<MovieSearchResult>>(result.results);
         }
 
-        public async Task<List<MovieSearchResult>> Upcoming(string langageCode)
+        public async Task<List<MovieSearchResult>> Upcoming(string langCode)
         {
-            var request = new Request($"movie/upcoming", BaseUri, HttpMethod.Get);
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("language", langageCode);
+            var request = new Request($"discover/movie", BaseUri, HttpMethod.Get);
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("language", langCode);
+            request.AddQueryString("with_release_type", "2|3");
+            var startDate = DateTime.Today.AddDays(7);
+            request.AddQueryString("release_date.gte", startDate.ToString("yyyy-MM-dd"));
+            request.AddQueryString("release_date.lte", startDate.AddDays(17).ToString("yyyy-MM-dd"));
```

Comment:
Can you please explain what is going on with the dates?

Reply:
@tidusjar I did the best I could to emulate `movie/upcoming` here, but the date logic for that endpoint is unclear. `movie/upcoming` sends back an object like this, but doesn't specify how it arrived at the values:

```json
"dates": {
    "maximum": "2016-09-22",
    "minimum": "2016-09-01"
}
```

Therefore, I had to assume the minimum and maximum were a certain offset from today. I compared the results I got with the results from `movie/upcoming` and they were the same for now. I will add comments to make this more clear.